src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -255,12 +256,15 @@ private async Task HandleClusterTransaction(DocumentsOperationContext context, M
             var array = new DynamicJsonArray();
             if (clusterTransactionCommand.DatabaseCommandsCount > 0)
             {
-                ClusterTransactionCompletionResult reply;
-                using (var timeout = new CancellationTokenSource(ServerStore.Engine.OperationTimeout))
-                using (var cts = CancellationTokenSource.CreateLinkedTokenSource(timeout.Token, HttpContext.RequestAborted))
+                var timeout = Database.ForTestingPurposes?.WaitForDatabaseResultsTimeoutInClusterTransaction?.Invoke(command) ??
```

Comment:
@karmeli87 @ayende 
Why do we need a timeout here in the first place?
At this point, the command was confirmed to be accepted and it was applied on this node, we are just waiting for it to be executed.
If you have a lot of those commands or the IO is slow this can take some time.

Reply:
Basically, the idea is for the client to know the documents were modified on the node.
If you call something like refresh you can get an old version - https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/how-to/refresh-entity#session-how-to-refresh-an-entity.
The second goal is to get the result documents - but that we handled in sharding so we can use the same logic.

Reply:
Shouldn't this be controlled from Client API?